pkg/generator/secret_template_reconciler.go

```diff
@@ -75,7 +103,21 @@ func (r *SecretTemplateReconciler) Reconcile(ctx context.Context, request reconc
 		return reconcile.Result{}, err
 	}
 
-	if secretTemplate.DeletionTimestamp != nil {
+	if secretTemplate.DeletionTimestamp.IsZero() {
+		if !controllerutil.ContainsFinalizer(&secretTemplate, finalizerName) {
```

Comment:
any way we can avoid finalizer here? without finalizer i guess we could lose "events" for deletion of secrets/secrettemplate...

Reply:
This was really to ensure we tidy up the tracking resources when a SecretTemplate is Deleted.

Thinking about this though I guess what we could do is if we get a Reconcile event that results in a 404 for SecretTemplate, we can clear that Templates tracking resources.

Reply:
nice!